src/Core/Security/ProviderConfiguration/Infrastructure/Repository/DbReadConfigurationRepository.php

```diff
@@ -0,0 +1,250 @@
+                    : null;
+                $jsonDecoded['authorization_rules'] =
+                    $this->readOpenIdConfigurationRepository->getAuthorizationRulesByConfigurationId(
+                        $configuration->getId()
+                    );
+
+                return new OpenIdCustomConfiguration($jsonDecoded);
+            case Provider::WEB_SSO:
+                $jsonSchemaValidatorFile = __DIR__ . '/../WebSSO/Repository/CustomConfigurationSchema.json';
+                $json = $configuration->getJsonCustomConfiguration();
+                $this->validateJsonRecord($json, $jsonSchemaValidatorFile);
+                $json = json_decode($json, true);
+
+                return new WebSSOCustomConfiguration(
+                    $json['trusted_client_addresses'],
+                    $json['blacklist_client_addresses'],
+                    $json['login_header_attribute'],
+                    $json['pattern_matching_login'],
+                    $json['pattern_replace_login']
+                );
+            default:
+                throw new \Exception("unknown configuration name, can't load custom config");
+        }
+    }
+
+    /**
+     * @param string $providerName
+     * @return Configuration
+     */
+    private function loadConfigurationByName(string $providerName): Configuration
+    {
+        $query = $this->translateDbName(
+            sprintf("SELECT *
+                FROM `:db`.`provider_configuration`
+                WHERE `name` = '%s'", $providerName)
+        );
+
+        $statement = $this->db->query($query);
+        $result = $statement->fetch(\PDO::FETCH_ASSOC);
+
+        return new Configuration(
+            (int)$result['id'],
+            $result['name'],
+            $result['type'],
+            $result['custom_configuration'],
+            (bool)$result['is_active'],
+            (bool)$result['is_forced']
+        );
+    }
+
+    /**
+     * @param int $id
+     * @return Configuration
+     */
+    private function loadConfigurationById(int $id): Configuration
+    {
+        $query = $this->translateDbName(
+            sprintf("SELECT *
+                FROM `:db`.`provider_configuration`
+                WHERE `id` = %d", $id)
+        );
+
+        $statement = $this->db->query($query);
+        $result = $statement->fetch(\PDO::FETCH_ASSOC);
+
+        return new Configuration(
+            (int)$result['id'],
+            $result['name'],
+            $result['type'],
+            $result['custom_configuration'],
+            (bool)$result['is_active'],
+            (bool)$result['is_forced']
+        );
+    }
+
+
+    /**
+     * @return array|Configuration[]
+     * @throws \Throwable
+     */
+    public function findConfigurations(): array
+    {
+        $configurations = [];
+        $query = $this->translateDbName("SELECT name FROM `:db`.`provider_configuration` where name <> 'web-sso'");
```

Comment:
just need a confirmation it's normal to exclude web-sso

Reply:
No, now it's fixed